From 6cad6a3a7e6103de24b89546ba43771e91994aa4 Mon Sep 17 00:00:00 2001
From: Nic Costa <nic.costa@gmail.com>
Date: Mon, 10 Jun 2019 23:15:28 -0500
Subject: [PATCH] fix compile errors mbed-os-5.13.0-rc2

fix compile errors when enabling the BG96 AT Cellular driver
through the Nordic SDK15
---
 .../stack/controller/sources/ble/bb/bb_ble_reslist.c  |  4 ++--
 .../stack/controller/sources/ble/lctr/lctr_act_enc.c  |  2 +-
 .../controller/sources/ble/ll/ll_main_enc_slave.c     |  2 +-
 .../TARGET_NRF5x/stack/sources/pal_bb_ble_rf.c        |  2 +-
 .../TARGET_MCU_NRF52840/config/sdk_config.h           | 10 +++++-----
 .../TARGET_NRF5x/TARGET_NRF52/serial_api.c            | 11 ++++++++---
 .../integration/nrfx/legacy/nrf_drv_rng.c             |  4 ++--
 .../modules/nrfx/drivers/include/nrfx_uart.h          |  1 +
 8 files changed, 21 insertions(+), 15 deletions(-)

diff --git a/features/FEATURE_BLE/targets/TARGET_CORDIO_LL/stack/controller/sources/ble/bb/bb_ble_reslist.c b/features/FEATURE_BLE/targets/TARGET_CORDIO_LL/stack/controller/sources/ble/bb/bb_ble_reslist.c
index 2a4933bd01..4da6eb99c4 100644
--- a/features/FEATURE_BLE/targets/TARGET_CORDIO_LL/stack/controller/sources/ble/bb/bb_ble_reslist.c
+++ b/features/FEATURE_BLE/targets/TARGET_CORDIO_LL/stack/controller/sources/ble/bb/bb_ble_reslist.c
@@ -25,8 +25,8 @@
 #include "bb_ble_api.h"
 #include "bb_ble_api_reslist.h"
 #include "bb_ble_api_pdufilt.h"
-#include "pal_bb_ble.h"
-#include "pal_crypto.h"
+#include "stack/platform/include/pal_bb_ble.h"
+#include "stack/platform/include/pal_crypto.h"
 #include "wsf_assert.h"
 #include "ll_math.h"
 #include "util/bda.h"
diff --git a/features/FEATURE_BLE/targets/TARGET_CORDIO_LL/stack/controller/sources/ble/lctr/lctr_act_enc.c b/features/FEATURE_BLE/targets/TARGET_CORDIO_LL/stack/controller/sources/ble/lctr/lctr_act_enc.c
index dd06f1b4ea..8aabf6cf05 100644
--- a/features/FEATURE_BLE/targets/TARGET_CORDIO_LL/stack/controller/sources/ble/lctr/lctr_act_enc.c
+++ b/features/FEATURE_BLE/targets/TARGET_CORDIO_LL/stack/controller/sources/ble/lctr/lctr_act_enc.c
@@ -27,7 +27,7 @@
 #include "wsf_msg.h"
 #include "wsf_trace.h"
 #include "util/bstream.h"
-#include "pal_crypto.h"
+#include "stack/platform/include/pal_crypto.h"
 #include <string.h>
 
 /*************************************************************************************************/
diff --git a/features/FEATURE_BLE/targets/TARGET_CORDIO_LL/stack/controller/sources/ble/ll/ll_main_enc_slave.c b/features/FEATURE_BLE/targets/TARGET_CORDIO_LL/stack/controller/sources/ble/ll/ll_main_enc_slave.c
index d47e843fb9..3f3c837a9d 100644
--- a/features/FEATURE_BLE/targets/TARGET_CORDIO_LL/stack/controller/sources/ble/ll/ll_main_enc_slave.c
+++ b/features/FEATURE_BLE/targets/TARGET_CORDIO_LL/stack/controller/sources/ble/ll/ll_main_enc_slave.c
@@ -26,7 +26,7 @@
 #include "ll_math.h"
 #include "wsf_msg.h"
 #include "wsf_trace.h"
-#include "pal_crypto.h"
+#include "stack/platform/include/pal_crypto.h"
 #include <string.h>
 
 /*************************************************************************************************/
diff --git a/features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/stack/sources/pal_bb_ble_rf.c b/features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/stack/sources/pal_bb_ble_rf.c
index a12adce13f..0dfc6f5907 100644
--- a/features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/stack/sources/pal_bb_ble_rf.c
+++ b/features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/stack/sources/pal_bb_ble_rf.c
@@ -21,7 +21,7 @@
  */
 /*************************************************************************************************/
 
-#include "pal_types.h"
+#include "stack/platform/include/pal_types.h"
 
 /**************************************************************************************************
   Macros
diff --git a/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52840/config/sdk_config.h b/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52840/config/sdk_config.h
index 57ec3c5ca0..37077b4b65 100644
--- a/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52840/config/sdk_config.h
+++ b/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/TARGET_MCU_NRF52840/config/sdk_config.h
@@ -1047,7 +1047,7 @@
 // <i> The nRF HW backend provide access to RNG peripheral in nRF5x devices.
 //==========================================================
 #ifndef NRF_CRYPTO_BACKEND_NRF_HW_RNG_ENABLED
-#define NRF_CRYPTO_BACKEND_NRF_HW_RNG_ENABLED 0
+#define NRF_CRYPTO_BACKEND_NRF_HW_RNG_ENABLED 1
 #endif
 // <q> NRF_CRYPTO_BACKEND_NRF_HW_RNG_MBEDTLS_CTR_DRBG_ENABLED  - Enable mbed TLS CTR-DRBG algorithm.
  
@@ -2921,7 +2921,7 @@
 // <e> NRFX_RNG_ENABLED - nrfx_rng - RNG peripheral driver
 //==========================================================
 #ifndef NRFX_RNG_ENABLED
-#define NRFX_RNG_ENABLED 0
+#define NRFX_RNG_ENABLED 1
 #endif
 // <q> NRFX_RNG_CONFIG_ERROR_CORRECTION  - Error correction
  
@@ -4953,7 +4953,7 @@
 // <e> RNG_ENABLED - nrf_drv_rng - RNG peripheral driver - legacy layer
 //==========================================================
 #ifndef RNG_ENABLED
-#define RNG_ENABLED 0
+#define RNG_ENABLED 1
 #endif
 // <q> RNG_CONFIG_ERROR_CORRECTION  - Error correction
  
@@ -5615,7 +5615,7 @@
 // <e> UART1_ENABLED - Enable UART1 instance
 //==========================================================
 #ifndef UART1_ENABLED
-#define UART1_ENABLED 0
+#define UART1_ENABLED 1
 #endif
 // </e>
 
@@ -6848,7 +6848,7 @@
 // <e> NRF_QUEUE_ENABLED - nrf_queue - Queue module
 //==========================================================
 #ifndef NRF_QUEUE_ENABLED
-#define NRF_QUEUE_ENABLED 0
+#define NRF_QUEUE_ENABLED 1
 #endif
 // <q> NRF_QUEUE_CLI_CMDS  - Enable CLI commands specific to the module
  
diff --git a/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/serial_api.c b/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/serial_api.c
index 089318ce01..07c20f6c2e 100644
--- a/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/serial_api.c
+++ b/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_NRF52/serial_api.c
@@ -194,6 +194,11 @@ NRF_ATFIFO_DEF(nordic_nrf5_uart_fifo_1, uint8_t, UART1_FIFO_BUFFER_SIZE);
 static uint8_t nordic_nrf5_uart_swi_mask_tx_0 = 0;
 static uint8_t nordic_nrf5_uart_swi_mask_rx_0 = 0;
 
+#if UART1_ENABLED
+static uint8_t nordic_nrf5_uart_swi_mask_tx_1 = 0;
+static uint8_t nordic_nrf5_uart_swi_mask_rx_1 = 0;
+#endif
+
 /**
  * Global variables expected by mbed_retarget.cpp for STDOUT.
  */
@@ -881,7 +886,7 @@ void serial_init(serial_t *obj, PinName tx, PinName rx)
         nordic_nrf5_uart_state[1].owner = NULL;
 
         /* Allocate a PPI channel for flow control */
-        ret = nrf_drv_ppi_channel_alloc(&nordic_nrf5_uart_state[1].ppi_rts);
+        ret = nrfx_ppi_channel_alloc(&nordic_nrf5_uart_state[1].ppi_rts);
         MBED_ASSERT(ret == NRF_SUCCESS);
 
         /* Clear RTS */
@@ -891,8 +896,8 @@ void serial_init(serial_t *obj, PinName tx, PinName rx)
         nrf_uarte_int_disable(nordic_nrf5_uart_register[1], 0xFFFFFFFF);
 
         NVIC_SetVector(UARTE1_IRQn, (uint32_t) nordic_nrf5_uart1_handler);
-        NRFX_IRQ_PRIORITY_SET(nrfx_get_irq_number(UARTE1_IRQn), APP_IRQ_PRIORITY_HIGHEST);
-        NRFX_IRQ_ENABLE(nrfx_get_irq_number(UARTE1_IRQn));
+        NRFX_IRQ_PRIORITY_SET(UARTE1_IRQn, APP_IRQ_PRIORITY_HIGHEST);
+        NRFX_IRQ_ENABLE(UARTE1_IRQn);
 #endif
     }
 
diff --git a/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_SDK_15_0/integration/nrfx/legacy/nrf_drv_rng.c b/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_SDK_15_0/integration/nrfx/legacy/nrf_drv_rng.c
index a3aed4bf86..0c71dda509 100644
--- a/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_SDK_15_0/integration/nrfx/legacy/nrf_drv_rng.c
+++ b/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_SDK_15_0/integration/nrfx/legacy/nrf_drv_rng.c
@@ -38,7 +38,7 @@
  * 
  */
 #include "sdk_common.h"
-#if NRF_MODULE_ENABLED(RNG)
+//#if NRF_MODULE_ENABLED(RNG)
 
 #include <stdint.h>
 #include <stddef.h>
@@ -280,4 +280,4 @@ NRF_SDH_STATE_OBSERVER(m_sd_state_observer, RNG_CONFIG_STATE_OBSERVER_PRIO) =
 
 #endif // SOFTDEVICE_PRESENT
 
-#endif // NRF_MODULE_ENABLED(RNG)
+//#endif // NRF_MODULE_ENABLED(RNG)
diff --git a/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_SDK_15_0/modules/nrfx/drivers/include/nrfx_uart.h b/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_SDK_15_0/modules/nrfx/drivers/include/nrfx_uart.h
index d55beb7247..9fcb7c51c5 100644
--- a/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_SDK_15_0/modules/nrfx/drivers/include/nrfx_uart.h
+++ b/targets/TARGET_NORDIC/TARGET_NRF5x/TARGET_SDK_15_0/modules/nrfx/drivers/include/nrfx_uart.h
@@ -68,6 +68,7 @@ enum {
 #if NRFX_CHECK(NRFX_UART0_ENABLED)
     NRFX_UART0_INST_IDX,
 #endif
+    ENUM_PAD,
     NRFX_UART_ENABLED_COUNT
 };
 
-- 
2.21.0

